src/main/frontend/state.cljs

```diff
@@ -312,22 +313,28 @@
   which are merged."
   [& configs]
   (apply merge-with
-    (fn merge-config [current new]
-      (if (and (map? current) (map? new))
-        (merge current new)
-        new))
-    configs))
+         (fn merge-config [current new]
+           (if (and (map? current) (map? new))
+             (merge current new)
+             new))
+         configs))
 
 (defn get-config
   "User config for the given repo or current repo if none given. All config fetching
 should be done through this fn in order to get global config and config defaults"
   ([]
    (get-config (get-current-repo)))
   ([repo-url]
-   (merge-configs
-    default-config
-    (get-in @state [:config ::global-config])
-    (get-in @state [:config repo-url]))))
+   (try
+     (merge-configs
+      default-config
+      (get-in @state [:config ::global-config])
+      (get-in @state [:config repo-url]))
+     (catch :default e
+       (do
+         (log/error "Cannot parse config files" e)
+         (log/error "Restore default config")
+         default-config)))))
```

Comment:
If we only set default-config the user loses all of their graph config which would cause annoying, silent bugs e.g. certain properties aren't rendered correctly or their journals start saving in a different format

Reply:
Can we keep the valid config from the global one and the current graph's config as a temporal fix?

Reply:
We don't have access to both valid configs here. Since global config doesn't have validation, it's probably the global config being invalid that causes the failure. We could add the current graph config as a temporal fix but it doesn't guarantee that there still isn't an issue and leads to another issue - a poor user notification experience. These catch's are occurring in a low level fn that can be called a number of times during render, which leads to notifying the user a number of times, at a time potentially much later than when they edited the invalid config. If we don't want illegal global config we should just validate and tell the user when they have entered it

Reply:
What about backup the user's last valid config on exit?